In [1]:
!pip install pymysql


[notice] A new release of pip is available: 23.2 -> 24.0
[notice] To update, run: C:\Users\juana\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


In [2]:
import pymysql

In [13]:


# Establecer la conexión
conn = pymysql.connect(
    host="localhost", # o la dirección IP de tu servidor MySQL
    port=3306, # puerto por defecto de MySQL
    user="root", # tu usuario de MySQL
    passwd="DGDAdfsdff##44@", # tu contraseña de MySQL
    db="p2", # el nombre de tu base de datos
    charset="utf8" # codificación para evitar problemas con caracteres especiales
)

# Crear un cursor
cursor = conn.cursor()
#cursor = conexion.cursor(dictionary=True) esto da error 



In [16]:
# Ejecutar una consulta SQL para obtener los nombres de las columnas
sql_query = """
SELECT Categoria, ValorTotalVentas, ProductosVendidos, PorcentajeValorTotal, PorcentajeProductosVendidos
FROM p2hwm3;
"""

# Ejecutar la consulta
cursor.execute(sql_query)

# Obtener los resultados
resultados = cursor.fetchall()

# Imprimir los resultados
for fila in resultados:
    print(fila)

# Cerrar el cursor y la conexión
cursor.close()
conn.close()


('Bikes', 94651172.70469621, 90268, 86.17, 32.83)
('Components', 11802593.286429755, 49044, 10.74, 17.84)
('Clothing', 2120542.524800972, 73670, 1.93, 26.8)
('Accessories', 1272072.8839257574, 61932, 1.16, 22.53)


Los resultados que has mostrado representan los datos obtenidos de la consulta SQL realizada a la tabla "p2hwm3" en la base de datos "p2". Cada fila corresponde a un registro en la tabla y contiene los valores de las columnas especificadas en la consulta.

Aquí está la traducción de los resultados:

- Categoría: Bicicletas
  - Valor Total de Ventas: 94,651,172.70
  - Productos Vendidos: 90,268
  - Porcentaje del Valor Total: 86.17%
  - Porcentaje de Productos Vendidos: 32.83%

- Categoría: Componentes
  - Valor Total de Ventas: 11,802,593.29
  - Productos Vendidos: 49,044
  - Porcentaje del Valor Total: 10.74%
  - Porcentaje de Productos Vendidos: 17.84%

- Categoría: Ropa
  - Valor Total de Ventas: 2,120,542.52
  - Productos Vendidos: 73,670
  - Porcentaje del Valor Total: 1.93%
  - Porcentaje de Productos Vendidos: 26.8%

- Categoría: Accesorios
  - Valor Total de Ventas: 1,272,072.88
  - Productos Vendidos: 61,932
  - Porcentaje del Valor Total: 1.16%
  - Porcentaje de Productos Vendidos: 22.53%

Estos resultados proporcionan información sobre las ventas y la distribución de productos en diferentes categorías de la tienda, mostrando los valores totales de ventas, la cantidad de productos vendidos y los porcentajes respectivos. Si necesitas más análisis o tienes alguna pregunta adicional sobre estos resultados, no dudes en preguntar. Estoy aquí para ayudarte.

subir la info a https://github.com/OAVallejos/sqlpractica